adrs/20241029-versioning.md

```diff
@@ -0,0 +1,68 @@
+# Versioning Scheme Tied to FAST Releases
+
+**authors:** [Ludo](https://github.com/ludoo),  [Julio](https://github.com/jccb), [Simone](https://github.com/sruffilli) \
+**date:** Oct 29, 2024
+
+## Status
+
+Proposed
+
+## Context
+
+Our current versioning scheme releases new versions based on changes across all modules. This approach was suitable when modules were the primary focus of development. However, with the increasing importance of FAST, this process no longer aligns with our priorities. We need a versioning scheme that reflects the significance of FAST releases and allows for more frequent updates to modules and documentation. The current release process wasn't designed with FAST in mind, causing friction and delaying releases.
+
+## Proposal
+
+Tie major version releases to FAST releases containing breaking changes or new architectural paradigms. Minor version releases will be used for module updates and documentation changes.
+
+### Development Workflow:
+
+* **Modules and Documentation:** Changes to modules and documentation will be made directly to the `master` branch.
+* **FAST Development:** FAST development will occur in a dedicated, protected branch named `fast-dev`. All changes to `fast-dev` must be submitted via Pull Requests..
+
+### FAST Release Process:
+
+1. Merge `master` into `fast-dev`. This ensures that the latest module and documentation changes are included in the FAST release.
+1. Merge all pending PRs into `fast-dev`.
```

Comment:
```suggestion
1. Merge all pending PRs into `master`.
1. Merge `master` into `fast-dev`. This ensures that the latest module and documentation changes are included in the FAST release.
```
Changing target PR branch requires interaction from contributor, I think this way it will be easier

Reply:
We discussed the whole approach last night with Julio and this needs to be simplified, there's a diagram he made but I guess he had not ime to update this yet.

Reply:
Updated. @wiktorn can you take another look.